iam_tags.go

```diff
@@ -98,3 +93,50 @@ func SetTagsDiff(_ context.Context, diff *schema.ResourceDiff, meta interface{})
 
 	return nil
 }
+
+func getExternalyManagedTags(roleTags TagMap, ignoredTags IgnoreTags) TagMap {
+	externalTags := TagMap{}
+	//Loop Through ignored keys and ignored key prefixes, checking if a tag exists that should be ignored
+	for k := range ignoredTags.Keys {
+		if val, ok := roleTags[k]; ok {
+			externalTags[k] = val.(string)
+		}
+	}
+
+	for p := range ignoredTags.KeyPrefixes {
+		for k, v := range roleTags {
+			if strings.HasPrefix(k, p) {
+				externalTags[k] = v.(string)
+			}
+		}
+	}
+	return externalTags
+}
+
+//Combine two tag maps.  Values in map2 will overwrite values in map1 if they exist in both maps
+func combineTagMaps(map1 TagMap, map2 TagMap) TagMap {
+	LocalMap := TagMap{}
+
+	for k, v := range map1 {
+		LocalMap[k] = v
+	}
+	for k, v := range map2 {
+		LocalMap[k] = v
+	}
+
+	return LocalMap
+}
+
+func removeIgnoredTags(allTags TagMap, ignoredTags IgnoreTags) TagMap {
+	for k := range allTags {
+		if _, ok := ignoredTags.Keys[k]; ok {
+			delete(allTags, k)
+		}
+		for kp := range ignoredTags.KeyPrefixes {
+			if strings.HasPrefix(k, kp) {
+				delete(allTags, k)
+			}
+		}
+	}
+	return allTags
+}
```

Comment:
Also I noticed this function does not have any associated tests

Reply:
Sorry I brought you down that rabbit hole. Since that map is passed by reference, should we ditch the return statement? It's probably a bit pedantic but I don't see a need for it

Reply:
Nah I meant I spent an hour debugging it when I first wrote it.  Had about a dozen print statements running before I realized my map was getting modified by this function.  I'll check and see if I can remove the return statement.

Reply:
I think I'm going to have it create a new map inside the function and return that to keep with functional programming.  I really don't like the map getting modified within a function.  You cool with that?

Reply:
That's the right move. keep it pure like the function call above it